mathesar/views/api.py

```diff
@@ -292,3 +301,50 @@ def create(self, request):
         serializer = DataFileSerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
         return create_datafile(request, serializer.validated_data)
+
+
+class ConstraintViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin, CreateModelMixin, DestroyModelMixin):
+    serializer_class = ConstraintSerializer
+    pagination_class = DefaultLimitOffsetPagination
+
+    def get_queryset(self):
+        return Constraint.objects.filter(table__id=self.kwargs['table_pk']).order_by('-created_at')
+
+    def create(self, request, table_pk=None):
+        table = get_table_or_404(table_pk)
+        serializer = ConstraintSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        # If we don't do this, the request.data QueryDict will only return the last column's name
+        # if there are multiple columns.
+        data = request.data.dict()
+        data['columns'] = request.data.getlist('columns')
```

Comment:
When I try to submit POST requests via the DRF-created web ui, `request.data` seems to come through as a `dict` type here, which breaks these lines.  Am I doing something wrong?  The request I submitted was:
```
{
    "name": "aconstraint",
    "type": "unique",
    "columns": ["Status"]
}
```
The error I got was 
![image](https://user-images.githubusercontent.com/6829750/127875647-7aaeb9d7-d795-4f48-8551-bfb7bcf727ee.png)

Fixing that error just let the next line cause the error (since there's no `getlist` method of python `dict` either).

Reply:
This should be fixed now. I don't know why it's a `QueryDict` sometimes and a dict other times, but both cases are handled.